File: src/bss/bss_status.rs
use num_enum::TryFromPrimitive;

/// The status of a BSS.
/// Based on nl80211_bss_status from linux/include/uapi/linux/nl80211.h
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, TryFromPrimitive)]
#[repr(u32)]
pub enum BssStatus {
    Authenticated = 0,
    Associated,
    IbssJoined,
    NotAssociated = u32::MAX,
}

File: dotnet/Kawaiifi.Net/BssStatus.cs
using System.Runtime.Versioning;

namespace Kawaiifi.Net;

/// <summary>The status of a BSS.</summary>
[SupportedOSPlatform("linux")]
public enum BssStatus
{
    /// <summary>The local station is authenticated with the BSS.</summary>
    Authenticated,
    /// <summary>The local station is associated with the BSS.</summary>
    Associated,
    /// <summary>The local station has joined the IBSS.</summary>
    IbssJoined,
}
